### PR TITLE
Isolate Sinopia server operations to a single module

### PR DESCRIPTION
**Pull request by mjgiarlo**
This is a refactoring commit with the following pieces included:
* Remove knowledge of Sinopia server internals, including authentication, from `ImportResourceTemplate` component
* Extract `sinopiaServerSpoof` module from `sinopiaServer` to disentangle them
* Drop unnecessary `.js[x]` extension from `import` statements for consistency

### CODE REVIEW

File: __tests__/sinopiaServerSpoof.test.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import Config from '../src/Config'
```
Comment (jermnelson) [inline __tests__/sinopiaServerSpoof.test.js]: Needs copyright statement.

File: src/sinopiaServerSpoof.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+const barcodeRt = require('../static/spoofedFilesFromServer/fromSinopiaServer/resourceTemplates/Barcode.json')
```
Comment (jermnelson) [inline src/sinopiaServerSpoof.js]: Needs copyright statement.

File: __tests__/sinopiaServerSpoof.test.js (hunk 1 above)
Comment (mjgiarlo) [reply __tests__/sinopiaServerSpoof.test.js]: Added.

File: src/sinopiaServerSpoof.js (hunk 2 above)
Comment (mjgiarlo) [reply src/sinopiaServerSpoof.js]: Added.

**Comment by mjgiarlo** [comment]
@jermnelson I've added the missing copyright statement to the files touched in my branch earlier *and* all the other files that were missing it. I also corrected the package.json so that it asserts the right license.

**Pull request merged**